File: 21-merge-two-sorted-lists/merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    
    def mergeTwoLists(self, list1, list2):
        """
        :type list1: Optional[ListNode]
        :type list2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        head=ListNode(0)
        tail=head
        while list1!=None and list2!=None:
            if(list1.val<list2.val):
                node=ListNode(list1.val)
                tail.next=node
                tail=tail.next
                list1=list1.next
            else:
                node=ListNode(list2.val)
                tail.next=node
                tail=tail.next
                list2=list2.next
        
        while list1!=None:
            node=ListNode(list1.val)
            tail.next=node
            tail=tail.next
            list1=list1.next

        while list2!=None:
            node=ListNode(list2.val)
            tail.next=node
            tail=tail.next
            list2=list2.next
        return head.next
            


        

File: 21-merge-two-sorted-lists/merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    void insert(ListNode* &head,int value){
        ListNode* node=new ListNode(value);
        if(head==NULL){
            head=node;
            return;
        }

        ListNode* temp=head;
        while(temp->next!=NULL){
            temp=temp->next;
        }
        temp->next=node;

    }
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode* head=NULL;
        if(list1==NULL && list2==NULL){
            return head;
        }if(list1==NULL) return list2;
        if(list2==NULL)  return list1;
        while(list1!=NULL && list2!=NULL){
            if(list1->val<list2->val){
                insert(head,list1->val);
                list1=list1->next;

            }else{
                insert(head,list2->val);
                list2=list2->next;
            }
        }
        while(list1!=NULL){
            insert(head,list1->val);
            list1=list1->next;
        } 
        while(list2!=NULL){
            insert(head,list2->val);
            list2=list2->next;
        }
        return head;
    }
};